### DISCUSSION TITLE
docker buildx create behaviour changed since 0.9.0

### DISCUSSION THREAD

**Discussion by cchafer** [docker/buildx#1358] [Q&A]
Hi, 
I was using  a teamcity build agent image with buildx capabilities.
This image is used to publish some docker images to remote repositories.

Until version 0.8.2 (included), this simple Dockerfile was working just fine:
```
FROM jetbrains/teamcity-agent:2022.04.1-linux-sudo
COPY --from=docker/buildx-bin:0.8.2 /buildx /usr/libexec/docker/cli-plugins/docker-buildx
RUN docker buildx install \
 && docker buildx create --name multi-arch-builder --use 
```

_(I made it shorter, the real dockerfile also add arguments through a buildkitd.toml file)._

Since version 0.9.0, this dockerfile doesn't work anymore, because `docker buildx create` now tries to connect to docker.sock.
And obviously, inside a docker build task, I don't have access to the sock.
```
#7 0.322 ERROR: failed to initialize builder multi-arch-builder (multi-arch-builder0): Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

Am I missing something, or is it juste some side effect due to issue #1206  ?

I'm quite confident this is not an issue, so I'm asking here, please, tell me if this message is in the wrong place.

Thank you

**Comment by jedevc** [comment]
This sounds suspiciously similar to https://github.com/docker/buildx/issues/1293 to me...

If you change your script to have an explicit `--driver docker-container` for the `docker buildx create` command, does it keep working as before?

**Reply by cchafer** [reply]
Just tried, it doesn't seem to change anything
```
  > [stage-0 4/4] RUN docker buildx create --name multi-arch-builder --use --driver docker-container:
#8 0.273 ERROR: failed to initialize builder multi-arch-builder (multi-arch-builder0): Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

**Comment by jedevc** [comment]
Sorry, have worked out what the actual issue is - it looks like a combination of #887 and #1206 which causes an error when contacting the docker API on builder creation. Ideally though, we shouldn't need to call the docker API here at all, will take a look at it soon :)
